__tests__/app.test.js

```diff
@@ -0,0 +1,43 @@
+const request = require("supertest")
+const app = require("../app.js")
+const db = require ("../db/connection.js")
+const seed = require("../db/seeds/seed.js")
+const data = require("../db/data/test-data")
+
+beforeEach(()=>{
+    return seed(data)
+})
+
+afterAll(()=>{
+    db.end()
+})
+
+describe("all bad URLs",()=>{
+    describe("/api/topics",()=>{
+        test("404 URL NOT FOUND", () => {
+        return request(app)
+        .get("/api/tipics")
+        .expect(404)
+        .then(({ body }) => {
+        expect(body.msg).toBe("URL NOT FOUND")
+        }) 
+     })
+    })
+})
+
+describe('/api/topics', () => {
+    test("200: responds with an array of topic objects", () => {
+        return request(app)
+        .get ("/api/topics")
+        .expect(200)
+        .then(({body:{topics}}) => {
+            topics.forEach((topic) => {
+                expect(topic).toMatchObject({
+                    slug: expect.any(String),
+                    description: expect.any(String),
+                })
+            })
+        })   
```

Comment:
If the `topics` array is empty, the `forEach` loop will be skipped and your test assertions will be skipped along with it, consider adding a check for the expected length of the array before the loop to ensure it isn't empty